MovingBlocks/Terasology
# feature(subsystem-prepare): extract Network and R&R code from EventSystemImpl

### Contains

Preparement to extraction EntitySystem: extract Network's replication and Recording of R&R from EventSystem.
Used Decorators for compose this together.

### How to test

EventSystem loads:

1. Start TS
2. Create/Load Game
3. Check that loads and works (movement, windows opens, blocks breaks)

Network:

1. Start TS
2. Start server
3. Join to Server
4. Check that loads and works (your events should replicated to server)

R&R:

1. Start TS
2. Create/Load game 
3. record 
4. replay
5. check that works.

Ref PR: https://github.com/MovingBlocks/Terasology/pull/4323
